src/mlpack/methods/ann/layer/randomized_relu_impl.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file methods/ann/layer/randomized_relu_impl.hpp
+ * @author Shubham Agrawal
+ *
+ * @code
+ * @article{Bing Xu2015,
+ *  author = {Bing Xu, Naiyan Wang, Tianqi Chen, Mu Li},
+ *  title = {Empirical Evaluation of Rectified Activations in Convolutional 
+ *      Network},
+ *  year = {2015},
+ *  url = {https://arxiv.org/pdf/1505.00853.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RRELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RRELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "randomized_relu.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+RReLU<InputDataType, OutputDataType>::RReLU(
+    const double lowerbound,
+    const double higherbound):
+    lowerbound(lowerbound),
+    higherbound(higherbound),
+    deterministic(false),
+    alpha(0.0)
+{
+  if (lowerbound < 1.0 || higherbound < 1.0)
+  {
+    Log::Fatal << "Lowerbound and higherbound must be greater than 1."
+        << std::endl;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void RReLU<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  if (deterministic)
+  {
+    alpha = 2.0 / (higherbound + lowerbound);
+  }
+  else
+  {
+    alpha = 1.0 / math::Random(lowerbound, higherbound);
+  }
```

Comment:
nit : 
```
   alpha = 2.0 / (higherbound + lowerbound);
    if (!deterministic)
    {
      alpha = 1.0 / math::Random(lowerbound, higherbound);
    }
```

Reply:
I thought the former approach was neater than you suggested for explanation and assembly optimization point of view.

Reply:
Umm, okay. Just not a huge fan of if-else statements. I always prefer removing else or using ternary operator. Upto you, left it as nit comment anyway.